lib/pal/PAL.cpp

```diff
@@ -96,9 +96,18 @@ namespace PAL_NS_BEGIN {
         bool isLoggingInited = false;
 
 #ifdef HAVE_MAT_LOGGING
-        std::recursive_mutex          debugLogMutex;
-        std::string                   debugLogPath;
-        std::unique_ptr<std::fstream> debugLogStream;
+
+        std::recursive_mutex& GetDebugLogMutex()
+        {
+            static std::recursive_mutex debugLogMutex;
+            return debugLogMutex;
+        }
+        
+        std::unique_ptr<std::fstream>& GetDebugLogStream() noexcept
+        {
+            static std::unique_ptr<std::fstream> debugLogStream;
```

Comment:
Globals-Bad argument aside, is having an uninitialized `unique_ptr` really that expensive that we need to have a static unique_ptr that we pass around as ref? The global ptr won't matter until it is initialized, so, why add this method?

Reply:
I did not pick these globals arbitrarily. I am not trying to make a generic statement that globals are bad. I fixed only places where the code was crashing on iOS if I tried to make OneDS initialization calls in +load method before OneDS globals are constructed